airflow/contrib/operators/aws_athena_operator.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.aws_athena_hook import AWSAthenaHook
+
+
+class AWSAthenaOperator(BaseOperator):
+    """
+    Airflow operator to run presto queries on athena.
+
+    :param query: Presto to be run on athena. (templated)
+    :type query: str
+    :param database: Database to select. (templated)
+    :type database: str
+    :param output_location: s3 path to write the query results into. (templated)
+    :type output_location: str
+    :param aws_conn_id: aws connection to use.
+    :type aws_conn_id: str
+    """
+
+    ui_color = '#44b5e2'
+    template_fields = ('query', 'database', 'output_location')
+
+    @apply_defaults
+    def __init__(self, query, database, output_location, aws_conn_id='aws_default', *args, **kwargs):
+        super(AWSAthenaOperator, self).__init__(*args, **kwargs)
+        self.query = query
+        self.database = database
+        self.output_location = output_location
+        self.aws_conn_id = aws_conn_id
+        self.client_request_token = kwargs.get('client_request_token')
+        self.query_execution_context = kwargs.get('query_execution_context') or {}
+        self.result_configuration = kwargs.get('result_configuration') or {}
```

Comment:
All these three kwargs needs to be named parameters so that we don't end up passing unknown parameters to the BaseOperator constructor.

And the defaults should be None, not `{}` (as mutable defaults in a function are bad in Python, so this line would become

```suggestion
        self.result_configuration = result_configuration or {}
```

Reply:
Not necessarily, result_configuration and query_context have few keys which are built from other parameters like database, output_location etc.., We either have to use params passed to boto3 function(which requires user to know about above params) or with current approach user have to pass obvious parameters like database and output_location so that he don't have to worry about specifics(for beginner level users).

Reply:
The problem here is that kwargs might contain `result_configuration` -  and this will cause the following warning to be emitted:

```
Invalid arguments were passed to AWSAtheneaOperator. Support for passing such arguments will be dropped in Airflow 2.0. Invalid arguments were: args=[], kwargs={'result_configuration':...}
```

In short, if an operator understands a arg/kwarg it must not be passed to the super constructor (currently on line 50).

Reply:
I will move them to args with defaults values.

Reply:
Importantly they should not have default of `{}` in the function - see https://docs.python-guide.org/writing/gotchas/